syntax = "proto3";

package layout;

import "google/protobuf/struct.proto";

service SiVistaLayout {
  rpc LayoutGen (LayoutGenRequest) returns (stream SiVistaResponse);
  rpc Hyperexpressivity (HyperexpressivityRequest) returns (stream SiVistaResponse);
  rpc GetVersion (Empty) returns (SiVistaResponse);
  rpc GetFile (FilePathRequest) returns (SiVistaResponse);
}

message LayoutGenRequest {
  string netlist_data = 1;
  string job_id = 2;
  string cell = 3;
  string s3_prefix = 4;
  string tech_json = 5;
  map<string, google.protobuf.Value> params = 6;
}

message SiVistaResponse {
  string status = 1;
  string message = 2;
}

message HyperexpressivityRequest {
  string gds_folder_path = 1;
  string job_id = 2;
  string cell = 3;
  string s3_prefix = 4;
  string netlist_data = 5;
  string tech_json = 6;
  map<string, google.protobuf.Value> params = 7;
}

message Empty {}

message FilePathRequest {
  string file_path = 1;
}